File: 프로그래머스/0/120876. 겹치는 선분의 길이/겹치는 선분의 길이.py
def solution(lines):
    answer = 0
    count = [0] * 200  # -100부터 100까지 총 200칸을 저장하기 위한 배열

    # 각 선분의 범위만큼 카운트 증가
    for line in lines:
        for i in range(line[0], line[1]):  # 시작점부터 끝점-1까지
            count[i + 100] += 1  # -100을 보정하여 배열의 인덱스를 맞춤

    # 두 개 이상의 선분이 겹친 부분의 개수 계산
    answer += count.count(2)  # 두 개 이상 겹친 부분
    answer += count.count(3)  # 세 개 이상 겹친 부분

    return answer

# ✅ 예제 테스트 실행
print(solution([[0, 1], [2, 5], [3, 9]]))  # 결과값: 2
print(solution([[-1, 1], [1, 3], [3, 9]]))  # 결과값: 0
print(solution([[0, 5], [3, 9], [1, 10]]))  # 결과값: 8

File: 프로그래머스/0/120876. 겹치는 선분의 길이/겹치는 선분의 길이.java
import java.util.*;

public class Solution {
    public static int solution(int[][] lines) {
        int answer = 0;
        int[] count = new int[200];  // -100부터 100까지 총 200칸

        // 각 선분의 범위만큼 카운트 증가
        for (int[] line : lines) {
            for (int i = line[0]; i < line[1]; i++) {
                count[i + 100] += 1;  // 인덱스 보정
            }
        }

        // 두 개 이상의 선분이 겹친 부분의 개수 계산
        for (int c : count) {
            if (c == 2 || c == 3) {
                answer += 1;
            }
        }

        return answer;
    }

    // ✅ 예제 테스트 실행
    public static void main(String[] args) {
        System.out.println(solution(new int[][]{{0, 1}, {2, 5}, {3, 9}}));   // 결과: 2
        System.out.println(solution(new int[][]{{-1, 1}, {1, 3}, {3, 9}})); // 결과: 0
        System.out.println(solution(new int[][]{{0, 5}, {3, 9}, {1, 10}})); // 결과: 8
    }
}
